tests/metadata/test_genconverter.py

```diff
@@ -59,6 +59,96 @@ def test_simple_roundtrip_defaults(attr_and_vals, strat):
     assert inst == converter.structure(converter.unstructure(inst), cl)
 
 
+@given(simple_typed_classes() | simple_typed_dataclasses(), unstructure_strats)
+def test_simple_roundtrip_with_extra_keys_forbidden(cls_and_vals, strat):
+    """
+    Simple classes can be unstructured and restructured with forbid_extra_keys=True.
+    """
+    converter = Converter(unstruct_strat=strat, forbid_extra_keys=True)
+    cl, vals = cls_and_vals
+    inst = cl(*vals)
+    unstructured = converter.unstructure(inst)
+    assert "Hyp" not in repr(unstructured)
+    assert inst == converter.structure(unstructured, cl)
+
+
+@given(simple_typed_attrs(defaults=True), unstructure_strats)
+def test_simple_roundtrip_defaults_with_extra_keys_forbidden(
```

Comment:
I think what I had in mind at the time was to make sure that it hadn't somehow made it error in the wrong order (i.e. when there are keys present in the structured object that aren't present in the dict), but given the current implementation I think that's not really a concern, and `test_forbid_extra_keys_nested_override` would probably catch that in a simple case anyway, so I've gone ahead and deleted.  Definitely nice to avoid having too many tests with hypothesis - can really increase the testing time.